ast/ast.go

```diff
@@ -292,6 +294,15 @@ func (ScalarSchemaType) isSchemaType() {}
 func (SizedSchemaType) isSchemaType()  {}
 func (ArraySchemaType) isSchemaType()  {}
 
+// IndexAlternation represents ALTER INDEX action.
+type IndexAlternation interface {
```

Comment:
While implementing this, I noticed that what might typically be named as `Alteration` is named as `Alternation` in the existing code. For consistency, I continued with the name `Alternation` in my changes. However, it might be worth reviewing this naming for clarity in future updates.

Reply:
It sounds good. I would like you to leave them as `Alternation` for now and fix them after merging this PR.

Reply:
OK, I'll raise another PR.